step-automation-packages/step-automation-packages-controller/src/main/java/step/automation/packages/AutomationPackageExecutionPlugin.java

```diff
@@ -0,0 +1,36 @@
+package step.automation.packages;
+
+import step.core.execution.ExecutionContext;
+import step.core.plugins.IgnoreDuringAutoDiscovery;
+import step.core.plugins.Plugin;
+import step.engine.plugins.AbstractExecutionEnginePlugin;
+
+@IgnoreDuringAutoDiscovery
+@Plugin
+public class AutomationPackageExecutionPlugin extends AbstractExecutionEnginePlugin {
+
+    private final static String AUTOMATION_PACKAGE_LOCK = "$automation.package.lock";
+    private final AutomationPackageLocks automationPackageLocks;
+
+    public AutomationPackageExecutionPlugin(AutomationPackageLocks automationPackageLocks) {
+        this.automationPackageLocks = automationPackageLocks;
+    }
+
+    @Override
+    public void executionStart(ExecutionContext context) {
+        super.executionStart(context);
+        String automationPackageId = (String) context.getPlan().getCustomField(AutomationPackageEntity.AUTOMATION_PACKAGE_ID);
```

Comment:
not sure if the plan is always set at this stage especially for external repositories (ALM, Jira, etc)

Reply:
Reviewed the sequence of calls which is done in ExecutionEngineRunner.execute, the executionStart hook is only called if the plan is not null (this is all done after the import action and import hooks)

Reply:
Still added a check on plan and will only lock if it's set.